### ISSUE TITLE
Products are synced and visible in inventory but do not appear in shop

### ISSUE THREAD

**Issue by Copterview** [facebookincubator/facebook-for-woocommerce#134]
No matter how ofter I repeat, with private of business account, new pixels or old ones, since I deleted the upload once, it never show the option to push the synced products into the shop. Also there is no info how to move products from inventory to shop either.
The tick box shown in video, show products in shop, does not appear in the 1.6.5

**Comment by dmitridr** [comment]
Hello,

Thanks for writing in. It would help to have a link to the video you are referring to, or a screenshot of what you see with an arrow pointing to the missing element.

Assuming you are talking about the shops option, it may be the case that you or your page are blacklisted from using shops, in this case the tick box (we call it a slider) to move items into the shop is not shown. This can happen if you are in one of the following countries : North Korea, Cuba, Iran, Syria. 

Unfortunately this is the policy of the shops team, and there is not much we can do to change that. If you are not in one of those countries and you are experiencing this, please give us some information about your page (link to page or page id) and we can investigate why you are in the blacklist or maybe there is some other bug.

**Comment by Copterview** [comment]
Dear Dimitri,

Thanks for your response.
I am in Germany and my page (privately managed) does have an shop. It is simply empty. When I activate the tool it syncs the products and I can click the link to check if the upload worked. 

I am happy to send a screenshot, but I have no idea what to show. 

Here is my Facebook page
http://prntscr.com/h5lx6o <http://prntscr.com/h5lx6o>
The only product is added manually. Currently managed privately, I also tied with the business manager with no success.

Here is my Woocommerce shop on Wordpress
http://prntscr.com/h5lxje <http://prntscr.com/h5lxje>

And the products are successfully sync in a fb catalogue
http://prntscr.com/h5mfnk <http://prntscr.com/h5mfnk>

But as said, the shop on my page stay empty there is nil info how to link the assets tot he shop.

Hope the helps for an analysis.

Best

Michael


COPTERview airDATA UG
Behrenstrasse 28
D-10117 Berlin

Mobil:0171 30 31 193
Tel:030 120 53 365
Fax:030 233 20 785

www.copterview.eu

**Comment by jcnieves** [comment]
Im having this same problem also and created a new issue for this but seems it is common.
Is there a way to fix this? I have removed the store, the collections, re added the pixels too, recyncd but nothing, what is happening on the facebook side that is not allowing this?
Thanks.

**Comment by micb11** [comment]
I also have the same problem. For some reason it is showing 6 out of 421 products. If i update a product manually in woocommerce it eventually will appear in the facebook shop. But i do not want to manually sync 421 products

**Comment by dmitridr** [comment]
Hi @Copterview unfortunately I checked our internal blacklist, and it seems that your page had been blacklisted by our shops team at some past point in time, I am not sure why. I've whitelisted your page since you clearly have a self-serve shop already. 

Please try to go through setup again, be sure to set the shop selector to 'on' on the final step before hitting finish. You should see a warning that your existing shop will be overwritten.

**Comment by dmitridr** [comment]
@jcnieves @micb11 Sounds like a completely different problem. This problem was one of the shop not even appearing at all or even being an option, due to our blacklist. Let's move the discussion of your problem into the other issue you that one of you mentioned.

**Comment by jcnieves** [comment]
Mine works fine now. Thanks.

**Comment by Copterview** [comment]
@dmitridr I used your jscript deletion tweak, reconfigured, re-installed the plugin and started all over, but unfortunately I can never reach the screen to set the shop selector. Is there a tipp how to reset the plugin, so that the shop selector appears again?
Don't get me wrong, all steps of the setup are ok, simply the shop selector does not appear anymore.

**Comment by tahoff** [comment]
@Copterview, are you still experiencing an issue? It looks like the shop might have been disabled on your end.

**Comment by Copterview** [comment]
Yes, it is a nightmare. Product are now in my shop on the management site. But the shop does not appear in the visitors view and since ten day most product show processing.

I din’t disable the shop. Shop button is on in templates. There is no other way as I know to manage the shops appearance. All other errors appeared after I started to use your tool. Sorry for that bad news.

But many functions are buggy since then. I.E. In my Facebook mobile app, my managed paged doesn’t appear anymore and when I switch to visitor view, the link to switch back to management view is broken. Didn’t expect Facebook being so buggy.

Michael

**Comment by micb11** [comment]
I managed to solve my problem by hiding the shop and then republishing it. After that the message went away. The plugin then started to process the products but after a while is stopped processing them. If I manually go into a product and update on the woocommerce site it will then appear on facebook. But there still seems to be a syncing issue.

**Comment by Copterview** [comment]
Just in case you still interested:



Notice
People cannot currently see this product because the seller's Page has not been approved to sell on Facebook.


Michael

**Comment by huebschie** [comment]
We are having this same issue. Only a few of our many products are showing on our Facebook shop. We'd very much like to link the two. Any thoughts? (www.facebook.com/ovrride/)

**Comment by Nibiko** [comment]
I have exactly same issue. My products are synced to Facebook. and Ive products catalog.
However my shop is empty. Any solutions?

**Comment by acmanbd** [comment]
I have a similar problem. It was working fine but all of a sudden it stopped working and our products are not syncing with facebook shop. Please let me know what to do?

**Comment by joanesle** [comment]
Looks like I'm having the very same problem.  Products don't sync.  At one point WC has assigned a facebook ID to the products, but they don't show up facebook.

**Comment by sammul** [comment]
@Nibiko @acmanbd @joanesle Can you guys send me the url link to your Facebook page so that I can check whether your shop is blocked or not?

**Comment by Nibiko** [comment]
https://www.facebook.com/catsvsdogsgiftshop/

Thanks

**Comment by sammul** [comment]
@Nibiko It turned out your issue is different than the one here: we suspect that you didn't toggle on shop feature during setting in the extension. Would you mind go through the setup again and make sure the shop option is on. Remember that doing this will override your current shop in your page, which seems setup by you manually.

**Comment by Nibiko** [comment]
Could you tell me where is the shop option setting in the extension? I
tried to delete plugin and re activated a couple time but nothing changed.
If you can screen shot, really appreciate.

**Comment by sammul** [comment]
@Nibiko Did you toggle on this?

![Uploading Screen Shot 2017-12-28 at 4.36.05 PM.png…]()

**Comment by Nibiko** [comment]
Hi I checked my setting and I don't have the button which add products to a facebook shop.

![0135ffff5cfd39cf4d527513c418bf859d8ff20b1b](https://user-images.githubusercontent.com/34614275/34435752-9effa0c8-ecdb-11e7-96e0-f2aacc306286.jpg)

what have i done wrong??

**Comment by adrenalinmototoys** [comment]
@Nibiko  and @sammul I do not have that button neither.  I have been passed back and forth bewtween FB help and woocommerce help each placing blame to the other.  THIS IS THE FIRST I have seen this button with the option to push products to the facebook shop tab and not just into my fb ad manager catalog like everyone else is complaining about here.  PLEASE tell me why we don't see that option?  Here is my page https://www.facebook.com/ADRMOTOdotCOM/ however the shop tab is empty and only visible on my page but my products show only in my ad manager section. THANKS!

**Comment by sammul** [comment]
@Nibiko the screenshot you sent is after setup, which is our control panel. What I sent in this thread is the screenshot for the initial setup. Would you mind to try to delete everything and set up again, and to see if the shop option comes up or not?
cc @adrenalinmototoys

**Comment by Nibiko** [comment]
I disactivated and deleted the plugin, restart the computer, re-download
the file and activated it again.
Automatically it synced to my facebook page and I can't go initial setup.
Could you tell me how to delete complately and re-setup initialy again pls?

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by Nibiko** [comment]
I disactivated and deleted the plugin, restart the computer, re-download
the file and activated it again.
Automatically it synced to my facebook page and I can't go initial setup.
Could you tell me how to delete complately and re-setup initialy again pls?




<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by dmitridr** [comment]
Hi Nibiko,

Thanks for your patience. Here's how you can `completely` reset the plugin.

1. Click `Reconfigure` on the configuration page.
2. In the popup, click 'Advanced Options'
3. Click the button that says `Delete`
4. Confirm

This will reset everything from scratch. Then, you should be able to set up from scratch. When you get to the final step, don't click `Finish` until you confirm you can see the slider in the screenshot that @sammul posted. It's easy to miss. 

Hope that helps.

**Comment by acmanbd** [comment]
My Facebook page is www.facebook.com/shahebbibi dotCom

Problem still persist.. Please help guys..

**Comment by acmanbd** [comment]
Www.Facebook.com/shahebbibidotcom

**Comment by Nibiko** [comment]
Finally its worked!! Thank you for your help, @sammul @dmitridr

**Comment by dmitridr** [comment]
@acmanbd your shop looks fine to me here https://www.intern.facebook.com/pg/ShahebBiBidotcom/shop/

If there's a problem with the product sync no longer working, please open a new issue. Also we changed how sync works in v1.8.0 so doing an upgrade and a reset may fix the problem.

**Comment by milosmili** [comment]
Simply to say the transfer from my Woo shop to my FB business page may last forever and you'll never be warned if any issues in transfer / blacklisted products (even they all respect FB guidelines). The plugin does not show any upload progress I have only 210 products with a total of 4,7K variations and after 3 days I'm on my 45th variations loaded into my FB business page (and of course not all variations are loaded). Simple calculation is if the "progress" continue with the same speed It will be over in 47 days!! And what if a shop has 10,000 products ????

**Comment by milosmili** [comment]
What is the speed of transfer from Woo to FB page sing the magic plugin ? 1 variation per 6 hours or 1 variation per 24 hours ? Thank you for your ever prompt reply.

**Comment by milosmili** [comment]
is this a bug in 1.8.5 ?

Using Chrome and while logged into my FB profile / page plenty of my products have: "Your product is being processed. This usually takes a few minutes. We'll notify you when processing is complete and your product is visible.". Therefore cannot proceed to checkout on my Web site

However on the same PC using Firefox, and NOT logged in to FB,  the same products do not carry the above message.

Pls advise.

Thx.

**Comment by milosmili** [comment]
Just updated to ver. 1.8.6 and it crashes my site with Error 500 when trying to UPDATE a product. FB ID for the product failed to be created.

Many using the plugin are frustrated as we get no adequate support from FB. WooCommerce also denies any responsibility and forward all our inquiries to FB

**Comment by grabrail** [comment]
Similar issue, using 1.8.7, Shop Syncs, but no products appear on my Facebook shop.

Website is www.abrv.co.uk
Facebook Store is facebook.com/ABRVClothing

Could I be checked on the white/Black list please.

**Comment by kryptex** [comment]
Is there any solution to this problem yet?

**Comment by VonTalavang** [comment]
looks like no one cares

**Comment by VonTalavang** [comment]
ok, following sammul instructions.
There's is a button now

**Comment by adilpasha123** [comment]
https://www.facebook.com/laylooopakistan/

i am trying to sync my woocommerce products to fb. but its failing

**Comment by adilpasha123** [comment]
please help

**Comment by leodelto** [comment]
I have the exactly same issue. My products are synced to Facebook from Woocommerce. On my Facebook Business Account, Business Settings => Data Sources => Catalogs => Products, I can se my products, not all but some of them. But on my Facebook page shop (http://bit.ly/2sE2J7D) is empty. Any solutions, please?

**Comment by leodelto** [comment]
Hi everyone. I have the exact same issue. My products are synced to Facebook from Woocommerce. From my Facebook Business Account I can see on Business Settings => Data Sources => Catalogs => Products that some of the products, not all of them are there. But on my Facebook Page Shop (http://bit.ly/2sE2J7D) is empty. Any solutions, please?

**Comment by leodelto** [comment]
SOLVED: I deleted the shop tab on my Facebook page => Settings => Edit Page => Log Out => Log In and putted back again. Went to Woocommerce => Settings => Integration and clicked on Force Resync. Now all the products are showing on my catalog and on my Facebook page.

If you want to try this solution I recommend that first go to your Facebook Business Account, Business Settings => Data Sources => Catalogs => Products and delete all the products there and only then follow the previous steps. I say this because I don't know if the Force Resync will duplicate the products or not.

Hope I helped!
